### PR TITLE
Improve the speed test calculations

### PR DESCRIPTION
**Pull request by mustafaboleken**
https://github.com/ant-media/conference-call-application/issues/327
https://github.com/ant-media/conference-call-application/issues/326

### CODE REVIEW

File: react/src/pages/AntMedia.js [hunk 1]
```diff
@@ -463,73 +464,119 @@ function AntMedia(props) {
...
-                console.log("* jitter: " + jitter);
-                console.log("* outgoingBitrate: " + outgoingBitrate);
-                console.log("* bandwidth: " + bandwidth);
-
-                let speedTestResult = {};
-
-                if (rtt >= 150 || packetLost >= 2.5 || jitter >= 80 || ((outgoingBitrate / 100) * 80) >= bandwidth) {
-                    console.log("-> Your connection quality is poor. You may experience interruptions");
-                    speedTestResult.message = "Your connection quality is poor. You may experience interruptions";
-                } else if (rtt >= 50 || packetLost >= 1 || jitter >= 30 || outgoingBitrate >= bandwidth) {
```
Comment (mekya) [inline react/src/pages/AntMedia.js:503]: Let's increase the RTT time to 100, packetLost percentage to 2 and jitter to 80

File: react/src/pages/AntMedia.js [hunk 2]
```diff
@@ -463,73 +464,119 @@ function AntMedia(props) {
...
-                    console.log("-> Your connection is fair, but you may experience interruptions");
-                    speedTestResult.message = "Your connection is fair, but you may experience interruptions";
```
Comment (mekya) [inline react/src/pages/AntMedia.js:505]: Change the message something like that 
Your connection is moderate; occasional disruptions may occur.

File: react/src/pages/AntMedia.js [hunk 3]
```diff
@@ -463,73 +464,119 @@ function AntMedia(props) {
...
-                } else {
```
Comment (mekya) [inline react/src/pages/AntMedia.js:506]: Add an extra level rtt >= 50 || packetLostPercentage >= 1 || jitter >= 30 with a different message 

Your connection is good!

File: react/src/pages/AntMedia.js [hunk 4]
```diff
@@ -463,73 +464,119 @@ function AntMedia(props) {
...
-                    console.log("-> Your connection is good");
-                    speedTestResult.message = "Your connection is optimal";
```
Comment (mekya) [inline react/src/pages/AntMedia.js:508]: Change the message

Your connection is great!

**Review by mekya — CHANGES_REQUESTED** [review]
Hi Mustafa,

I have suggestions
- Change logic from packetLost to pcaketLostpercentage
- Add extra level
- Change the messages to make it easy and user-friendly

Take a look at the comments for more details

Let me know if you have questions

File: react/src/pages/AntMedia.js [hunk 5]
```diff
@@ -463,73 +464,121 @@ function AntMedia(props) {
...
+        updatedStats.audioJitter = (parseFloat(statsList.current[statsList.current.length - 1].audioJitter) + parseFloat(statsList.current[statsList.current.length - 2].audioJitter))/2.0;
+        updatedStats.audioJitter = (updatedStats.audioJitter === -1) ? 0 : updatedStats.audioJitter;
+
+        updatedStats.currentOutgoingBitrate = parseInt(statsList.current[statsList.current.length - 1].currentOutgoingBitrate) // we can use the last value
+        updatedStats.currentOutgoingBitrate = (updatedStats.currentOutgoingBitrate === -1) ? 0 : updatedStats.currentOutgoingBitrate;
+
+        speedTestForPublishWebRtcAdaptor.current?.stop("speedTestStream" + speedTestStreamId.current);
+        let rtt = ((parseFloat(updatedStats.videoRoundTripTime) + parseFloat(updatedStats.audioRoundTripTime)) / 2).toPrecision(3);
+        let packetLost = parseInt(updatedStats.videoPacketsLost) + parseInt(updatedStats.audioPacketsLost);
+        let packetLostPercentage = (updatedStats.videoPacketsLost+updatedStats.audioPacketLost)/(statsList.current[statsList.current.length - 1].videoPacketsLost+statsList.current[statsList.current.length - 1].audioPacketsLost) * 100
```
Comment (mekya) [inline react/src/pages/AntMedia.js:545]: Let's talk what the packetLostPercentage is

**Review by mekya — CHANGES_REQUESTED** [review]
Let's talk about what the packetLostPercentage is

File: react/src/pages/AntMedia.js (hunk 5 above)
Comment (mustafaboleken) [reply react/src/pages/AntMedia.js:545]: Scheduled a meeting on tomorrow at 3 PM, after our discussion finishes, I will write meeting notes here. @mekya

**Pull request merged**